packages/core/src/discovery.js

```diff
@@ -94,6 +97,42 @@ function parseDomResources({ url, domSnapshot }) {
   }, new Map([[rootResource.url, rootResource]]));
 }
 
+async function parseImageSrcset({ imageLinks }, { resources, meta }, { allowedHostnames }) {
+  let log = logger('core:snapshot');
+  const promiseList = [];
+
+  imageLinks = imageLinks.filter((link) => hostnameMatches(allowedHostnames, link));
```

Comment:
will it cover picture, video, source tags as well?

Reply:
video tags are handled separately where we capture poster. poster can only be one. picture case is handled